app/controllers/admin/poll/polls_controller.rb

```diff
@@ -74,4 +75,13 @@ def load_search
       @search = search_params[:search]
     end
 
+    def resource
+      @poll = Poll.find(params[:id]) unless @poll
+      @poll
```

Comment:
Is this line necessary? As far as I know, Ruby always returns the value of the last statement but if you can provide some insight @microweb10, that'd be great :smile_cat:

Reply:
Well, it will not be necessary if you don't use `unless @poll`. But, what will happen if `@poll` was already defined? `@poll = Poll.find(params[:id])` will never be executed and the method `resource` can return something different to `@poll`. Am I wrong?

Reply:
Oh yes, you're absolutely right! Sorry about the inconvenience! :bowing_man:

Reply:
What about `@poll ||= Poll.find(params[:id])`?

Reply:
I thought about it @javierm but wouldn't that approach avoid fetching a poll with a different ID? :thinking:

Reply:
Mmm... I thought `@poll ||= Poll.find(params[:id])` was (almost) exactly equivalent to `Poll.find(params[:id]) unless @poll`, the only difference being it returned `@poll` if already defined. Is there something I'm missing :thinking:?

Reply:
Yeah, exactly the same. I thought about that, but I just wanted to use the same patterns I've seen in other controllers. I thought it was maybe more readable to other users. But, personally, I like it more. Changed!